tinkerbell.tf

```diff
@@ -0,0 +1,93 @@
+provider "tinkerbell" {
+  grpc_authority = "127.0.0.1:42113"
+  cert_url       = "http://127.0.0.1:42114/cert"
+}
+
+resource "tinkerbell_hardware" "foo" {
+  data = <<EOF
+{
+  "id": "b91059ff-f174-46e6-bbc9-fd3a789f883f",
+  "metadata": {
+    "facility": {
+      "facility_code": "sjc1",
+      "plan_slug": "c3.small.x86",
+      "plan_version_slug": ""
+    },
+    "instance": {},
+    "state": ""
+  },
+  "network": {
+    "interfaces": [
+      {
+        "dhcp": {
+          "arch": "x86_64",
+          "ip": {
+            "address": "192.168.1.5",
+            "gateway": "192.168.1.1",
+            "netmask": "255.255.255.248"
+          },
+          "mac": "${module.tinkerbell.worker_mac_addr[0]}",
+          "uefi": false
+        },
+        "netboot": {
+          "allow_pxe": true,
+          "allow_workflow": true
+        }
+      }
+    ]
+  }
+}
+EOF
+}
+
+resource "tinkerbell_template" "foo" {
+  name    = "foo"
+  content = <<EOF
+version: "0.1"
+name: Ubuntu_Focal
+global_timeout: 1800
+tasks:
+  - name: "os-installation"
+    worker: "{{.device_1}}"
+    volumes:
+      - /dev:/dev
+      - /dev/console:/dev/console
+      - /lib/firmware:/lib/firmware:ro
+    actions:
+      - name: "stream-ubuntu-image"
+        image: quay.io/tinkerbell-actions/image2disk:v1.0.0
+        timeout: 600
+        environment:
+          DEST_DISK: /dev/sda
+          IMG_URL: "http://192.168.1.1:8080/tink-ubuntu-2004-cloud-init.raw.gz"
```

Comment:
this is a custom image that would need to be replaced with something that could be fetched or built-in registry.tf